### ISSUE TITLE
MailSender not showing share window on Xiaomi Android 10

### ISSUE THREAD

**Issue by ElavanResu** [ACRA/acra#786]
I am using ACRA 5.5.1 in React Native 0.61.2. I am using Dailog + Mailer feature to send the crash reports.
`@AcraCore(buildConfigClass = BuildConfig.class,
    reportFormat = StringFormat.JSON)
@AcraMailSender(mailTo = "email",
   resSubject = R.string.acra_mail_subject,
   reportFileName = "ErrorReport",
   reportAsFile = true)
@AcraToast(resText = R.string.acra_toast_text)
public class MainApplication extends Application implements ReactApplication {

@Override
protected void attachBaseContext(Context base) {
    super.attachBaseContext(base);
    CoreConfigurationBuilder builder = new CoreConfigurationBuilder(this)
            .setBuildConfigClass(BuildConfig.class)
            .setReportFormat(StringFormat.JSON);
    builder.getPluginConfigurationBuilder(ToastConfigurationBuilder.class)
            .setResText(R.string.acra_toast_text);
   builder.getPluginConfigurationBuilder(MailSenderConfigurationBuilder.class)
           .setMailTo("email")
           .setResSubject(R.string.acra_mail_subject)
           .setReportFileName("ErrorReport")
           .setReportAsFile(true);
    ACRA.init(this, builder);
}}`

Now, when I am using the APK in android 5.0, everytime my app crashes it open up a small window at bottom to share the report via email. Phone is Sony Xperia E2362.

But when I am using the APK in Android 10, phone xiaomi poco f1. It doesn't show me any window to share the file. But right after I install my app, xiaomi scans the app and gives one option to open it. And when I open the app from that window I get that share report via email popup, after the app crashes. But not when I normally open my app from app menu. Bellow image shows the app scanning window which shows an option to open app and the second image which shows the share report via email when I use scanning window to open app.

https://i.stack.imgur.com/qXWPP.jpg

https://i.stack.imgur.com/II1BF.jpg

I don't know what am I missing. Please help.

**Comment by mikehardy** [comment]
Xiaomi is terrible, basically. Barely android and they break the APIs. I'm struggling with this on a different API (a couple actually) but if it works on everything else and not on Xiaomi I hate Xiaomi a little more and mentally mark whatever APIs I was using as "don't use if you want to support Xiaomi"

**Comment by F43nd1r** [comment]
I just confirmed this is working on a normal android 10 device.
I can't fix this issue without access to a device with the issue, sorry. I can accept a PR if you manage to fix it yourself.

**Comment by ElavanResu** [comment]
> I just confirmed this is working on a normal android 10 device.
> I can't fix this issue without access to a device with the issue, sorry. I can accept a PR if you manage to fix it yourself.

I think, mikehardy is right. I took some other Xiaomi phones from my neighbour and one samsung with android 9 and 10. Crash report worked in Samsung. But it failed in Xiaomi phones.

I got a log of Xiaomi poco f1 phone. Can you have a quick look in these logs and see if you can find anything? It would help me a lot. 
Thank you for your time.

Logs: https://gist.github.com/ElavanResu/eb45b52be299a313fdbc9794b31e2c07

**Comment by mikehardy** [comment]
you can also have a look at https://dontkillmyapp.com/xiaomi and see if you are able to effect things with settings changes

If you are, then you can detect the restrictions during normal app execution and open the settings panel without too much fuss - it's device-specific code which is never fun but the snippets are all over stackoverflow at least

I just got finished with another Xiaomi bug myself - they don't let you set TextField ActionCode handlers, so you can't help your users with contextual text actions. It's always something with them in my experience. Super popular though so :shrug:

**Comment by ElavanResu** [comment]
Ohhh, this actually works.
I also noticed the following line in logs

06-19 12:45:59.579  2032  2521 D ActivityTaskManagerServiceInjector: MIUILOG- Permission Denied Activity : Intent { act=android.intent.action.CHOOSER flg=0x10000000 cmp=android/com.android.internal.app.ChooserActivity (has extras) } pkg : com.agentnativeapp uid : 10811 tuid : 10049

At that point I knew it's because of some permissions getting restricted. So I gave permission to "Display pop-up windows while running in the background". ANd it worked. 
And your comment just confirmed that it's a permission problem. Thank you so much.

Now I have to just learn how to automatically inform user to give permissions when started the app on android.
